src/main/java/jaryl/duke/DataManager.java

```diff
@@ -0,0 +1,101 @@
+package jaryl.duke;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+/**
+ * DataManager handles reading/writing data from/to local disk storage
+ */
+public class DataManager {
+    private String filePath;
+    private static final String DELIMITER = " \\| ";
+
+    /**
+     * Constructor to instantiate a new DataManager
+     * @param filePath path to file
+     */
+    public DataManager(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Handles the creation and writing of data to file on local disk storage
+     * @param tasksList list of tasks
+     */
+    public void writeToFile(ArrayList<Task> tasksList) {
+        StringBuilder sb = new StringBuilder();
+        for(int i = 0; i < tasksList.size(); i++) {
+            sb.append(tasksList.get(i).fileFormat()).append("\n");
+        }
+
+        try {
+            BufferedWriter bw = new BufferedWriter(new FileWriter(this.filePath));
+            bw.write(sb.toString());
+            bw.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Handles the reading of data from existing file on local disk storage
+     */
+    public ArrayList<Task> readFromFile() throws DukeException {
```

Comment:
I personally think this method is slightly too long and has too many responsibilities. In order to follow **SLAP** closely, I think you can consider separate it into a few methods, including `createFile()`, `readFromFile()`, `createTasksFromFile()`, etc.